### PR TITLE
P2022-1510_Remove_user_assigned_to_the_board

### PR DESCRIPTION
**Pull request by msperski**

### CODE REVIEW

**Review by edych — APPROVED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 1]
```diff
@@ -184,4 +184,22 @@ public void delete(final int boardId, final String email) {
...
+        final Board board = boardRepository.findById(boardId)
+                .orElseThrow(() -> new NotFoundException("Board is not found"));
+
+        if (board.getCreator().equals(user)) {
+            throw new BadRequestException("User is the board owner.");
+        }
+        if(email.equals(user.getEmail()) || email.equals(board.getCreator().getEmail())){
+            board.getUsers().remove(user);
+        }else {
+            throw new BadRequestException("User can't delete board");
```
Comment (golldfish) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:202]: But why do you throw ‘User can’t delete board’? I believe this case is about removing user from the board and not the board itself.

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 2]
```diff
@@ -718,6 +715,86 @@ void getUserBoardsShouldReturnTwoBoardsIfUserCreatedOneAndIsAssignedToTheOtherOn
         assertThat(boards, hasSize(2));
     }
 
+    @Test
+    @DisplayName("removeAssignedUserShouldThrowNotFoundWhenUserOrBoardAreNotExist should throw Not Found")
+    void removeAssignedUserShouldThrowNotFoundWhenUserOrBoardAreNotExist(){
```
Comment (golldfish) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:720]: In this test you only check whether user does not exists, not the board. You can simplify this test:
`when(userRepository.findById(user.getUid())).thenReturn(Optional.empyty()));`

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 3]
```diff
@@ -718,6 +715,86 @@ void getUserBoardsShouldReturnTwoBoardsIfUserCreatedOneAndIsAssignedToTheOtherOn
...
+        when(userRepository.findById(user.getUid())).thenReturn(Optional.of(user));
+        when(boardRepository.findById(board.getId())).thenReturn(Optional.of(board));
+
+        //then
+        assertThrows(NotFoundException.class, () -> boardService.removeUserAssignedToTheBoard(user_.getUid(), board.getId(), user_.getEmail()));
+    }
+
+    @Test
+    @DisplayName("removeAssignedUserShouldThrowBadRequestWhenUserIsBoardOwner should throw Bad Request")
+    void removeAssignedUserShouldThrowBadRequestWhenUserIsBoardOwner(){
```
Comment (golldfish) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:737]: Isn't this test the same as removeAssignedUserShouldThrowBadRequestWhenBoardOwnerTriesToSelfDelete?

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 4]
```diff
@@ -718,6 +715,86 @@ void getUserBoardsShouldReturnTwoBoardsIfUserCreatedOneAndIsAssignedToTheOtherOn
...
+        final Board board = buildBoard(userOwner, EnumStateDto.CREATED, 10, Set.of());
+
+        //when
+        when(userRepository.findById(userOwner.getUid())).thenReturn(Optional.of(userOwner));
+        when(boardRepository.findById(board.getId())).thenReturn(Optional.of(board));
+
+        //then
+        assertFalse(board.getUsers().contains(user));
+    }
+
```
Comment (golldfish) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:797]: What with case when user is not a board owner and tried to self delete (when it's assigned to board)?

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 5]
```diff
@@ -727,4 +804,5 @@ private Board buildBoard(final User user, final EnumStateDto state, final int id
                 .users(users)
                 .build();
     }
-}
+
```
Comment (golldfish) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:807]: please remove empty line

**Review by golldfish — CHANGES_REQUESTED** [review]

File: src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java [hunk 6]
```diff
@@ -739,6 +736,88 @@ void getUserBoardsShouldReturnTwoBoardsIfUserCreatedOneAndIsAssignedToTheOtherOn
         assertThat(boards, hasSize(2));
     }
 
+    @Test
+    @DisplayName("removeAssignedUserShouldThrowNotFoundWhenUserNotExists should throw Not Found")
```
Comment (patrykzawadzkiintive) [inline src/test/java/com/intive/patronage22/szczecin/retroboard/service/BoardServiceTest.java:740]: If you using display name, then write display name in more redable waw. Ex. "Remove assigned user .."

Do that also with other methods.

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 7]
```diff
@@ -189,4 +189,23 @@ public void delete(final int boardId, final String email) {
...
+
+        final User user = userRepository.findById(uid)
+                .orElseThrow(() -> new NotFoundException("User is not found"));
+        final Board board = boardRepository.findById(boardId)
+                .orElseThrow(() -> new NotFoundException("Board is not found"));
+
+        if (board.getCreator().equals(user)) {
+            throw new BadRequestException("User is the board owner.");
+        }
+        if(email.equals(user.getEmail()) || email.equals(board.getCreator().getEmail())){
```
Comment (patrykzawadzkiintive) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:204]: if () space between if and the bracket

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 8]
```diff
@@ -189,4 +189,23 @@ public void delete(final int boardId, final String email) {
...
+            board.getUsers().remove(user);
+        }else {
+            throw new BadRequestException
```
Comment (patrykzawadzkiintive) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:207]: } else {

**Review by patrykzawadzkiintive — CHANGES_REQUESTED** [review]

**Review by patrykzawadzkiintive — APPROVED** [review]

**Review by MaciejJ0 — APPROVED** [review]

**Review by Radenzvous — APPROVED** [review]

**Review by Szachista — APPROVED** [review]

**Review by golldfish — APPROVED** [review]

File: src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java [hunk 9]
```diff
@@ -189,4 +189,23 @@ public void delete(final int boardId, final String email) {
...
+
+        final User user = userRepository.findById(uid)
+                .orElseThrow(() -> new NotFoundException("User is not found"));
+        final Board board = boardRepository.findById(boardId)
+                .orElseThrow(() -> new NotFoundException("Board is not found"));
+
+        if (board.getCreator().equals(user)) {
+            throw new BadRequestException("User is the board owner.");
+        }
+        if (email.equals(user.getEmail()) || email.equals(board.getCreator().getEmail())) {
```
Comment (daniel-kmiecik) [inline src/main/java/com/intive/patronage22/szczecin/retroboard/service/BoardService.java:204]: Are you sure that email is not null?

**Pull request merged**